includes/components/class-email.php

```diff
@@ -46,9 +46,55 @@ public function __construct( $args = [] ) {
 			add_action( 'post_updated', [ $this, 'set_email_defaults' ], 10, 3 );
 		}
 
+		// Add object specific tokens.
+		add_filter( 'hivepress/v1/emails/email/meta', [ $this, 'alter_email_meta' ] );
+
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Add object specific tokens.
+	 *
+	 * @param array $meta Email meta.
+	 * @return array
+	 */
+	public function alter_email_meta( $meta ) {
+		foreach ( hp\get_array_value( $meta, 'tokens', [] ) as $name => $args ) {
+
+			// Get model name.
+			$model_name = hp\get_array_value( (array) $args, 'model' );
```

Comment:
Please add an example of this API for any of the emails (e.g. "Listing Reported"), also make sure that filtering tokens and extending them doesn't affect other email functions.